includes/class-wp-job-manager-ajax.php

```diff
@@ -121,6 +121,11 @@ public function get_listings() {
 
 		$result['found_jobs'] = false;
 
+		// Give plugins the opportunity to control what JSON is returned
+		if ( $result = apply_filters( 'job_manager_ajax_get_jobs', null, $result, $jobs ) ) {
```

Comment:
This is a little confusing to me, I see 2 things happening here.

1) If not overridden, this will return null. This will make the conditional fail and it will move on.
2) In order to make this succeed, you are expected to define something like this:

```php
add_filter( 'job_manager_ajax_get_jobs', 'my_func_override_ajax_get_job`, 1, 3 );
my_func_override_ajax_get_job( $serves_no_purpose, $result, $jobs ) {
   return $result;
}
```

This seems overly complex and partially redundant.

Instead of using the `job_manager_ajax_get_jobs` filter to return the results and perhaps provide another avenue to change what is in the `$result` I think it would be better to maintain that functionality with the `job_manager_get_listings_result` filter, which is meant for this.

My suggestion is two-fold.

1) Rename the filter to be something more specific to the intent here, like: `job_manager_ajax_get_jobs_html_results`

then 

2) Change the definition to be more like this: `if ( true !== apply_filters( 'job_manager_ajax_get_jobs_html_results', '__return_true', $result, $jobs ) ) {`

This will allow you to still programmatically decide whether or not to override the results, but would also prevent result tampering from too many sources.

The resulting add_filter would then look more like this:

```php
add_filter( 'job_manager_ajax_get_jobs_html_results', 'my_func_override_ajax_get_job_html_results`, 1, 3 );
my_func_override_ajax_get_job_html_results( $should_return_html, $result, $jobs ) {
    if ( $result /* has something good in it */ ) {
        $should_return_html = false;
    }

    return $should_return_html;
}
```

Without this, it seems like we're adding more hard to understand filters into WPJM. One thing I'd like to do is start auditing and documenting the existing filters, so making sure new ones are well thought through and make sense is also a priority.